### ISSUE TITLE
Query planner unwraps interface implementations incorrectly

### ISSUE THREAD

**Issue by barthez** [apollographql/federation#817]
### Description

After updating to recent @apollo/gateway  package version we started observing and error when query on a field of type of an interface  cause `No such type [Type], so it can't be a fragment condition`. Simplified setup down below

### Example (actual behavior)

Service A:
```
type Query {
  myField: MyInterface
}

type MyInterface {
  name: String
}

type MyTypeA implements MyInterface {
  name: String
}

type MyTypeB implements MyInterface {
  name: String
}
```

Service B:
```
interface MyInterface {
 name: String
}

type MyTypeC implements MyInterface {
  name: String
}
```

The query:
```
query {
  myField: {
    ...MyIntFragment
  }
}

fragment MyIntFragment on MyInterface {
  name
}
```

result in such query to service A:
```
query {
  myField {
    ...on MyTypeA { name }
    ...on MyTypeB { name }
    ...on MyTypeC { name }
  }
}
```
which cause an error
```
Unknown type "MyTypeC".
```

### Expected behavior

Query to the service A doesn't include fragment for `MyTypeC` that isn't known for the service.

### Versions when problem occurs:

```
@apollo/federation@0.25.1
@apollo/gateway@0.28.3
@apollo/query-planner@0.2.1
```

### Versions when problem **did not** occur

```
@apollo/federation@0.20.0
@apollo/gateway@0.20.0
```

### Working example

https://github.com/barthez/query-planner-error-example

**Comment by barthez** [comment]
Hey @trevor-scheer, is there any chance you could give some feedback in here. I have tried to fix it with https://github.com/apollographql/federation/pull/816 but it turned out to break other things (when types are actually shared between services). If you could give me some hints I could try to fix it and create a PR.

**Comment by pcmanus** [comment]
The upcoming version of federation (2.0, currently in alpha) fixes this issue and I've pushed 41a9b7653ce3696dba40f903b8d5ac1fe967b2dc with a test for it. Fixing this is on the 0.x branch is unfortunately a bit tricky (federation 2 has a laregly rewritten query planner, and one of the reasons for such rewrite was to make it easier to solve that kind of issues).

**Comment by konalegi** [comment]
@pcmanus have you seen this PR? https://github.com/apollographql/federation/pull/864, It works for our cases, could you please take a look if makes sense.
